Source/scrollrt.cpp

```diff
@@ -640,7 +640,7 @@ static void DrawItem(const CelOutputBuffer &out, int x, int y, int sx, int sy, b
 		CelBlitOutlineTo(out, GetOutlineColor(*pItem, false), position, *cel, nCel);
 	}
 	CelClippedDrawLightTo(out, position, *cel, nCel);
-	if (pItem->_iAnimFrame == pItem->_iAnimLen)
+	if (pItem->_iAnimFrame == pItem->_iAnimLen || pItem->_iCurs == ICURS_MAGIC_ROCK)

```

Comment:
This seems like a bad way to fix it IMHO. Why is the magic rock special? What if someone introduces a new animated object? Will they also have to list the object here?

Reply:
yes

Reply:
Unfortunately the magic rock not having a drop animation and instead having a idle animation special to that item (see `ProcessItems()`). Of cause refactoring it in to a property that could be used more generally and would be nice, but fixing the issue under the current para dime is imo not degrading code quality.